From 098dfc89a7d38f83aa9de4d0fc05fb20865b0c01 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 07 Nov 2022 12:59:06 +0200
Subject: [PATCH] Revert "[SW-104919] gaudi2: security: fix for ppboot hang"

This reverts commit 6ec07f3f332501eb2f9b4580bcd8f79166a7b457.

Since the ARC stall hang issue observed in arcboot_rom's
memcmp() function called from rsa_verify() is root-caused and fix
is added to make sure ARC1 DCCM memory is not accessed from both
DMI and internal address, the changes introduced in this patch can be
reverted.

Change-Id: I6043fe83e0dcce9b6ab511b9c9afba08a0f3ea90
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 754743e..edc9c80 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -335,16 +335,13 @@
 	return 0;
 }
 
-/* TODO: This does not need to be global. Make it local when
- * https://jira.habana-labs.com/browse/SW-105874 is concluded.
- */
-static uint32_t magic;
 static int ppboot_bringup_preboot(u8 image_ptr)
 {
 	u32 image_size, payload_size, copy_size;
 	u32 flash_offset;
 	u32 num_cycles;
 	int ret;
+	uint32_t magic = 0;
 	uint8_t hash[SHA384_SUM_LEN] = { 0 };
 
 	flash_offset = ppboot_flash_image_offset(image_ptr);
